src/shared/resourcefetcher/utils.ts

```diff
@@ -0,0 +1,77 @@
+/*!
```

Comment:
"resourcefetcher" already has a dubious reason for existence, I don't think we should further build it by adding "resorucefetcher/utils". These functions could live in resourcefetcher.ts

Reply:
I'm a bit eh here, they use the fetchers but they're not necessarily "must use" functions for fetching operations. I feel like `resourcefetcher.ts` should encapsulate core functionalities/interfaces.

Reply:
resourcefetcher.ts is a mistake. The reason for putting this there is so that we have fewer files to cleanup later when the mistake is corrected.